android/src/main/java/io/sentry/capacitor/SentryCapacitor.java

```diff
@@ -326,21 +333,27 @@ public void clearBreadcrumbs() {
 
     @PluginMethod
     public void setExtra(PluginCall call) {
-        String key = call.getString("key");
-        String extra = call.getString("extra");
-        Sentry.configureScope(scope -> {
-            scope.setExtra(key, extra);
-        });
-        call.resolve();
+        if (call.getData().has("key") && call.getData().has("extra")) {
+            Sentry.configureScope(scope -> {
+                String key = call.getString("key");
+                String extra = call.getString("extra");
+                scope.setExtra(key, extra);
+            });
+
+            call.resolve();
+        }
     }
 
     @PluginMethod
     public void setTag(PluginCall call) {
-        String key = call.getString("key");
-        String value = call.getString("value");
-        Sentry.configureScope(scope -> {
-            scope.setTag(key, value);
-        });
-        call.resolve();
+        if (call.getData().has("key") && call.getData().has("value")) {
+            Sentry.configureScope(scope -> {
+                String key = call.getString("key");
+                String value = call.getString("value");
+                scope.setTag(key, value);
+            });
+
+            call.resolve();
+        }
```

Comment:
Same here! Does this `call.resolve()` need to be executed outside of the if statement?